### PR TITLE
fix(#2191): disregard floating windows when opening window on edit

### LINKED ISSUE #2191: Vim:E444: Cannot close last window
### Description

E5108: Error executing lua: ...cal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/view.lua:205: Vim:E444: Cannot close last window
stack traceback:
 [C]: in function 'nvim_win_close'
 ...cal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/view.lua:205: in function 'close'
 ...cal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/view.lua:214: in function 'close_this_tab_only'
 ...cal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/view.lua:227: in function 'close'
 ...y/nvim-tree.lua/lua/nvim-tree/actions/node/open-file.lua:348: in function 'fn'
 ...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:139: in function 'edit'
 ...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:149: in function 'f'
 ...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:32: in function <...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:30>

### Neovim version

```text
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 13.3.1 (a) 

### nvim-tree version

270c955

### Minimal config

```lua
{
      sync_root_with_cwd = true,
      respect_buf_cwd = true,
      renderer = {
        group_empty = true,
      },
      update_focused_file = {
        enable = true,
        update_root = true,
      },
      view = {
        adaptive_size = true,
        -- width = 40,
        float = {
          enable = false,
          open_win_config = {
            relative = "editor",
            border = "rounded",
            width = 40,
            height = 100000,
            row = 0,
            col = 0,
          },
        },
      },
      diagnostics = {
        enable = true,
      },
      actions = {
        open_file = {
          resize_window = true,
          quit_on_open = true,
        },
      },
      git = {
        ignore = false,
      },
    },
```

### Steps to reproduce

I enable "actions.open_file.quit_on_open"

reproduce steps:

1. enter a directory
2. new a file
3. open that file

if I disable the "quit_on_open", no this issue there.

### Expected behavior

_No response_

### Actual behavior

https://user-images.githubusercontent.com/8643866/236633300-7f64f541-1f52-4a8b-9f81-2e9618914d3c.mp4

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #2191

### CODE REVIEW

**Comment by TendonT52** [comment]
I've noticed that the recent change has caused unexpected behavior in my settings. According to the documentation, <CR> should initiate nvim-tree-api.node.open.edit(). However, when I press <CR>, it appears to execute nvim-tree-api.node.open.vertical() instead.

Here is my current configuration for your reference:
I'm using lazy.nvim

return {
  "nvim-tree/nvim-tree.lua",
  keys = {
    { "<leader>e", ":NvimTreeToggle<CR>" }
  },
  commit = "498e8793bbe73ab5235b3ee8f0aee32f5d01649f",
  init = function()
    local theme = require("config.theme")
    theme.highlight("NvimTreeCursorLine", { bg = theme.color.bg.floating.bright })
  end,
  opts = {
    disable_netrw = true,
    hijack_netrw = true,
    view = {
      number = true,
      relativenumber = true,
      float = {
        enable = true,
        quit_on_focus_loss = true,
        open_win_config = function()
          local screen_w = vim.opt.columns:get()
          local screen_h = vim.opt.lines:get() - vim.opt.cmdheight:get()
          local window_w = screen_w * 0.5
          local window_h = screen_h * 0.8
          local window_w_int = math.floor(window_w)
          local window_h_int = math.floor(window_h)
          local center_x = (screen_w - window_w) / 2
          local center_y = ((vim.opt.lines:get() - window_h) / 2)
              - vim.opt.cmdheight:get()
          return {
            border = "",
            relative = "editor",
            style = "minimal",
            row = center_y,
            col = center_x,
            width = window_w_int,
            height = window_h_int,
          }
        end,
      }
    },
    renderer = {
      add_trailing = false,
      group_empty = false,
      highlight_git = false,
      full_name = false,
      highlight_opened_files = "none",
      highlight_modified = "none",
      root_folder_label = ":~:s?$?/..?",
      indent_width = 4,
      indent_markers = {
        enable = false,
        inline_arrows = true,
        icons = {
          corner = "└",
          edge = "│",
          item = "│",
          bottom = "─",
          none = " ",
        },
      },
      icons = {
        webdev_colors = true,
        git_placement = "before",
        modified_placement = "after",
        padding = " ",
        symlink_arrow = " ➛ ",
        show = {
          file = true,
          folder = true,
          folder_arrow = true,
          git = false,
          modified = true,
        },
      },
      special_files = { "Cargo.toml", "Makefile", "README.md", "readme.md" },
      symlink_destination = true,
    },
  },
  config = function(_, opts)
    require("nvim-tree").setup(opts)
  end
}


I hope this issue can be resolved soon. Thank you.

**Comment by alex-courtis** [comment]
> I've noticed that the recent change has caused unexpected behavior in my settings. According to the documentation, CR should initiate nvim-tree-api.node.open.edit(). However, when I press CR, it appears to execute nvim-tree-api.node.open.vertical() instead.

I did not test floating windows. I'll see if I can come up with a fix for you to test.

**Comment by alex-courtis** [comment]
@TendonT52 please see #2212
